### PR TITLE
[usbdev,sival] Enable silicon for usbdev_aon_wake_reset, implement usbdev_aon_wake_disconnect

### LINKED ISSUE #19972: [chip-test,sival] chip_sw_usbdev_aon_wake_disconnect
### Test point name

chip_sw_usbdev_aon_wake_disconnect

### Host side component

Rust

### OpenTitanTool infrastructure implemented

Unknown

### Silicon Validation (SiVal)

Yes

### Emulation Targets

- [ ] None
- [X] CW310
- [X] Hyperdebug + CW310

### Contact person

alees24

### Checklist

Please fill out this checklist as items are completed. Link to PRs and issues as appropriate.
- [ ] Check if existing test covers most or all of this testpoint (if so, either extend said test to cover all points, or skip the next 3 checkboxes)
- [ ] Device-side (C) component developed
- [ ] Bazel build rules developed
- [ ] Host-side component developed
- [ ] HJSON test plan updated with test name (so it shows up in the dashboard)
- [ ] Test added to dvsim nightly regression (and passing at time of checking)
- [ ] For SiVal test cases, test is running relevant FPGA or silicon regression

> estimate 3

### PR DESCRIPTION
**Pull request by pamaury**
First two commits are backports.

Fixes #19972

### CODE REVIEW

**Comment by pamaury** [comment]
Should be ready for review now that the teacup backport has been done.

**Comment by pamaury** [comment]
I have implemented `usbdev_aon_wake_disconnect` and added it to this PR

File: sw/host/tests/chip/usb/usb.rs [hunk 1]
```diff
@@ -121,6 +123,38 @@ impl UsbOpts {
...
+    // Enable/disable VBUS.
+    pub fn enable_vbus(&self, transport: &TransportWrapper, en: bool) -> Result<()> {
+        // Enable VBUS sense on the board if necessary.
+        let Some(vbus_sense_en) = &self.vbus_sense_en else {
+            bail!("cannot control VBUS, you must specify --vbus-sense-en");
+        };
+        log::info!("{} VBUS sensing.", if en { "Enable" } else { "Disable" });
+        let vbus_sense_en_pin = transport.gpio_pin(vbus_sense_en)?;
+        vbus_sense_en_pin.write(en)?;
+        std::thread::sleep(Duration::from_millis(100));
```
Comment (nbdd0121) [inline sw/host/tests/chip/usb/usb.rs:144]: Perhaps add a comment about the reason?

**Review by nbdd0121 — APPROVED** [review]
LGTM, just a nit

**Pull request merged**